container_workflow_tool/distgit.py

```diff
@@ -439,22 +439,25 @@ def _clone_downstream(self, component, branch):
             self.logger.info("Using existing downstream repo: " + component)
             repo = Repo(component)
         else:
-            ccomponent = "container/" + component
-            self.logger.info("Cloning into: " + ccomponent)
+            hostname_url = u._get_hostname_url(self.conf)
             packager = u._get_packager(self.conf)
-            ret = subprocess.run([packager, "clone", ccomponent],
+            # if packager is fedpkg then namespace is `container` else `containers`
+            namespace = "container" if packager == "fedpkg" else "containers"
+            # If hostname_url is specified use `git` otherwise `fedpkg|rhpkg` command.
+            cmd = "git" if hostname_url else packager
+            component_path = f"{namespace}/{component}"
+            ccomponent = f"{hostname_url}/{component_path}.git" if hostname_url else component_path
```

Comment:
Just a nitpick over here - this seems a bit more readable to me:

```suggestion
            component_path = f"{namespace}/{component}"
            # If hostname_url is specified use `git` otherwise `packager` command.
            if hostname_url:
                cmd = "git"
                ccomponent = f"{hostname_url}/{component_path}.git"
            else:
                cmd = packager
                ccomponent = component_path
```